jobs/payment-jobs/tasks/eft_overpayment_notification_task.py

```diff
@@ -0,0 +1,128 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task to notify staff user for short name over payments."""
+from datetime import datetime, timedelta, timezone
+from typing import List
+
+from flask import current_app
+from pay_api.models import db
+from pay_api.models.eft_credit import EFTCredit as EFTCreditModel
+from pay_api.models.eft_short_name_links import EFTShortnameLinks as EFTShortnameLinkModel
+from pay_api.models.eft_short_names import EFTShortnames as EFTShortnameModel
+from pay_api.services.auth import get_emails_with_keycloak_role
+from pay_api.utils.enums import EFTShortnameStatus, Role
+from sentry_sdk import capture_message
+from sqlalchemy import and_, func
+
+from services.email_service import _render_eft_overpayment_template, send_email
+
+
+class EFTOverpaymentNotificationTask:  # pylint: disable=too-few-public-methods
+    """Task to notify staff qualified receivers.
+
+    Notify staff qualified receiver there is a pending unsettled amount on a short name due to overpayment or an account
+    has not been linked for thirty days.
+    """
+
+    date_override = None
+    short_names: dict = {}
+
+    @classmethod
+    def _get_base_query(cls):
```

Comment:
probably a better name for this? 

`get_shortnames_ids_with_credit_remaining` etc